### ISSUE TITLE
`SupervisedSqids` appears to not work as intended

### ISSUE THREAD

**Issue by dcrck** [sqids/sqids-elixir#32]
I followed the instructions in the [intro documentation](https://github.com/sqids/sqids-elixir?tab=readme-ov-file#convenience-place-context-under-your-supervision-tree) to add Sqids as a part of my supervision tree, but the example does not appear to work. Looking at the [supervisor documentation](https://hexdocs.pm/elixir/1.16.3/Supervisor.html#module-child_spec-1-function), I can see why:

> Finally, note it is also possible to simply pass the `Counter` module as a child:
> ```elixir
> children = [
>   Counter
> ]
> ```
> When only the module name is given, it is equivalent to `{Counter, []}`, which in our case would be invalid, which is why we always pass the initial counter explicitly.

So even if I pass `SupervisedSqids` as a child to my application supervisor, `child_spec/0` will never be called.

I was only able to get my own options working when I copied the contents of `__using__` into my module and removed `use Sqids`.

**Comment by g-andrade** [comment]
Thanks for raising this, I'm considering the possible fixes. People who expected it to be working may have their setup broken by a one (because the setup they actually intended will suddenly come into affect.)

**Comment by decius7bc** [comment]
Simple workaround until a proper fix gets published: Use the result of calling the module's `child_spec/0` function instead of just passing the module.

~~~elixir
defmodule MyApp.Application do
  # ...
  def start(_type, _args) do
     children = [
       MyApp.SupervisedSqids.child_spec(),
       # ...
     ]

     opts = [strategy: :one_for_one, name: MyApp.Supervisor]
     Supervisor.start_link(children, opts)
  end
end
~~~

**Comment by g-andrade** [comment]
* Changed docs to @decius7bc's workaround (it's really the best way);
*  0910060: Found a way to detect missing options (set in child_spec/0 but not used). Raise to force those affected into action: either keep what they have, or use the intended options but break compatibility with existing IDs;
* Otherwise, if options are the same but child_spec/0 is not called, log a warning.

Only thing left to do is adding test coverage of those cases.

**Comment by g-andrade** [comment]
Changes released in 0.2.0, pushed to Hex. Thanks all for the report & suggested workaround!
